packages/ui/src/components/ui/attachment/useAttachment.ts

```
@@ -9,24 +9,47 @@ export interface AttachmentProps {
+  canDeleteFile,
   onChange,
 }: AttachmentProps) => {
   const inputFileRef = useRef<HTMLInputElement>(null);
   const [files, setFiles] = useState<FileState[]>([]);
-  const fileUploadMutation = UsePostFileUpload();
+  const fileUploadMutation = UsePostFileUpload({
+    onSuccess: (res) => {
+      onChange(res.data?.ids || []);
+      setFiles((prev) =>
+        prev.map((file, index: number) => {
+          delete file.loading;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)